Utils/ObjectPool.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DUCK.Pooling
+{
+	public static class ObjectPool
```

Comment:
A few comments documenting how the pool works, when it should be used and any pitfalls would be good.

Part of those comments might be a some a warning against blind or excessive usage of a pool which could cause (assumption alert!) slow down of the GC or creating a large memory footprint and all the usual issues with premature optimisation.

Reply:
I tried to keep this relatively lightweight, so the pools aren't even created until you call ObjectPool.Instantiate<something> - this will create and maintain a pool for the objects to return to when soft-destroyed via ObjectPool.Destroy, but is otherwise transparently identical to the standard Instantiate and Destroy functions, in that they're called as fallbacks if there isn't a pool to use.

I will, though, add guards to prevent a pooled object being pooled a second time, and to prevent the pool from returning destroyed objects, since it's still possible to hard-Destroy something while it's in the pool.